src/core/server/saved_objects/migrations/actions/integration_tests/actions.test.ts

```diff
@@ -139,6 +149,69 @@ describe('migration actions', () => {
         })
       );
     });
+    it('resolves left with cluster routing allocation disabled', async () => {
+      expect.assertions(3);
+      await client.cluster.putSettings({
+        body: {
+          persistent: {
+            // Disable all routing allocation
+            cluster: { routing: { allocation: { enable: 'none' } } },
+          },
+        },
+      });
+      const task = initAction({
+        client,
+        indices: ['existing_index_with_docs'],
+      });
+      await expect(task()).resolves.toMatchInlineSnapshot(`
+              Object {
+                "_tag": "Left",
+                "left": Object {
+                  "type": "cluster_routing_allocation_disabled",
+                },
+              }
+            `);
+      await client.cluster.putSettings({
+        body: {
+          persistent: {
+            // Allow routing to existing primaries only
+            cluster: { routing: { allocation: { enable: 'primaries' } } },
```

Comment:
Adds tests for all (currently supported) settings that will case the migration to transition to a `FATAL` state.
I've only added tests for `persistent` settings because `transient` settings are [discouraged](https://www.elastic.co/guide/en/elasticsearch/reference/current/transient-settings-migration-guide.html#transient-settings-migration-guide)